sklearn/cluster/_hdbscan/hdbscan_.py

```diff
@@ -0,0 +1,1178 @@
+
+    max_cluster_size : int, default=0
+        A limit to the size of clusters returned by the `eom` cluster selection
+        algorithm. Has no effect if `cluster_selection_method=leaf`. Can be
+        overridden in rare cases by a high value for
+        `cluster_selection_epsilon`.
+
+    metric : str or callable, default='minkowski'
+        The metric to use when calculating distance between instances in a
+        feature array.
+
+        - If metric is a string or callable, it must be one of
+          the options allowed by `metrics.pairwise.pairwise_distances` for its
+          metric parameter.
+
+        - If metric is "precomputed", X is assumed to be a distance matrix and
+          must be square.
+
+    leaf_size : int, default=40
+        Leaf size for trees responsible for fast nearest
+        neighbour queries.
+
+    algorithm : str, default='best'
+        Exactly which algorithm to use; hdbscan has variants specialised
+        for different characteristics of the data. By default this is set
+        to `best` which chooses the "best" algorithm given the nature of
+        the data. You can force other options if you believe you know
+        better. Options are:
+        - `best`
+        - `generic`
+        - `prims_kdtree`
+        - `prims_balltree`
+        - `boruvka_kdtree`
+        - `boruvka_balltree`
+
+    memory : str, default=None
+        Used to cache the output of the computation of the tree.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    approx_min_span_tree : bool, default=True
+        Whether to accept an only approximate minimum spanning tree.
+        For some algorithms this can provide a significant speedup, but
+        the resulting clustering may be of marginally lower quality.
+        If you are willing to sacrifice speed for correctness you may want
+        to explore this; in general this should be left at the default True.
+
+    n_jobs : int, default=4
+        Number of parallel jobs to run in core distance computations (if
+        supported by the specific algorithm). For `n_jobs`
+        below -1, (n_cpus + 1 + n_jobs) are used.
+
+    cluster_selection_method : str, default='eom'
+        The method used to select clusters from the condensed tree. The
+        standard approach for HDBSCAN* is to use an Excess of Mass algorithm
+        to find the most persistent clusters. Alternatively you can instead
+        select the clusters at the leaves of the tree -- this provides the
+        most fine grained and homogeneous clusters. Options are:
+        - `eom`
+        - `leaf`
+
+    allow_single_cluster : bool, default=False
+        By default HDBSCAN* will not produce a single cluster. Setting this to
+        `True` will allow single cluster results in the case that you feel this
+        is a valid result for your dataset.
+
+    match_reference_implementation : bool, default=False
```

Comment:
I suspect we going to always want this as `False` so we can remove this parameter?